stdlib/src/builtin/string.mojo

```diff
@@ -925,13 +925,10 @@ struct String(
             UnsafePointer.address_of(self).bitcast[NoneType](),
         )
 
-    fn join[rank: Int](self, elems: StaticIntTuple[rank]) -> String:
+    fn join(self, *elems: Int) -> String:
```

Comment:
I think we should remove this after switching its call sites to use `Tuple`. Maybe just leave a TODO here.

Reply:
I just tried removing it, and it works, should it remove it now?

Also I just realized that `String.join` does not match the python implementation of `str.join`, in python only one argument is expected, not a sequence of arguments

Reply:
Let's save the refactoring of `join` for a different PR, please.